compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELowerToAIE.cpp

```diff
@@ -403,23 +403,79 @@ LogicalResult getStaticDims(Operation *op,
   return success();
 }
 
+/// Utility to get the static offsets, sizes and strides for
+/// `AIEX::NpuDmaMemcpyNdOp` with implicit addressing.
+LogicalResult getStaticDimsForImplicitAddressing(
+    Operation *op, MemRefType memrefType, SmallVectorImpl<int64_t> &staticSizes,
+    SmallVectorImpl<int64_t> &staticStrides) {
+  SmallVector<int64_t> shapeArr;
+  for (auto shape : memrefType.getShape()) shapeArr.push_back(shape);
+  int64_t sizeIndex = staticSizes.size() - 1;
+  for (int64_t i = shapeArr.size() - 1; i >= 0; i--) {
+    staticSizes[sizeIndex--] = shapeArr[i];
+  }
+  int64_t strideIndex = staticStrides.size() - 1;
+  // Since aiex.npu.dma_cpy_nd expects 3-dimensional stride attribute, it has
```

Comment:
Note the MLIR-AIE bump in this PR which will make the last dimension explicit: https://github.com/nod-ai/iree-amd-aie/pull/503

Reply:
Thanks for pointing that out!
I'll accordingly make changes to this revision (and the lit test).